### ISSUE TITLE
Max total files with Rolling

### ISSUE THREAD

**Issue by someone972** [NLog/NLog#4137]
I've tried pretty much everything I can think of, but can't seem to get this scenario to work. What I want to do is have a max total number of archive files, but also have rollover enabled. Preferably I'd also be able to control the max number of roll-over files per instance, but not required. I'm encoding the date into the log name each time it is run with ${longdate:cached=true} If I use Rolling as-is with the date encoded, it will keep a max of maxArchiveFiles in rollovers per instance, but will keep building up logs as I launch more instances. I want to control the total files in the folder, not the total per instance.

So for example, if I had the maxArchiveFiles set to 2, and let it roll over twice:
Log_2020-10-16 1048 (1).log
Log_2020-10-16 1048 (0).log
Log_2020-10-16 1048.log

And if I ran it again I'd like the older logs to roll off first, so after first start:
Log_2020-10-16 1048 (0).log
Log_2020-10-16 1048.log
Log_2020-10-16 1150.log

And if that instance then rolled over:
Log_2020-10-16 1048.log
Log_2020-10-16 1150 (0).log
Log_2020-10-16 1150.log

Hopefully this makes what I want to do clear, if not I can try and clarify. I'm fairly certain this isn't possible with the way the current archivers are set up, but want to confirm.

**Comment by snakefoot** [comment]
I'm guessing you are trying to use both dynamic and static archive-logic together. See also https://github.com/NLog/NLog/issues/3807#issuecomment-587661413 and https://github.com/NLog/NLog/issues/4105#issuecomment-685952654

If you followed the [issue-template](https://github.com/NLog/NLog/issues/new?template=question.md) and posted the content of your current `NLog.config`, then we could probably give you advice on how to make it work.

**Comment by someone972** [comment]
I would of course have posted the XML config if I still had it, but I had since overwritten it several times in my attempt to get it to work. Let me try and retrace my steps and see if I can get a few examples of XML that didn't work as I wanted it to.

**Comment by snakefoot** [comment]
If you have no opinion on how to make it work, then I guess this will work just fine:

```xml
<target type="file" name="logfile" filename="Log_${longdate:cached=true}.log" maxArchiveFiles="2" />
```

**Comment by someone972** [comment]
I had tried that before but found it was picking up the last part of the date and using that as the base for the sequence numbering. However, that gave me the idea of wrapping it in some other character, and it seems that when it doesn't end in a dot with some number it acts like I want it to. I think this meets my requirements of needing to keep an indication of which instance the logs belong to, rolling over logs when they exceed a set size, and keeping the total number of logs produced below a certain threshold. Looks like I got close before, but was missing the wrapping character to prevent the sequencing from getting messed up.
So this is what I ended up with:
```xml
<target type="file" name="logfile" filename="Log_[${longdate:cached=true}].log" maxArchiveFiles="2" archiveAboveSize="1000000" />
```

Thanks for your expertise, it was able to point me in the right direction.
